pydal/objects.py

```diff
@@ -100,7 +100,7 @@
 
 def csv_reader(utf8_data, dialect=csv.excel, encoding="utf-8", **kwargs):
     """like csv.reader but allows to specify an encoding, defaults to utf-8"""
-    csv_reader = csv.reader(TextIOWrapper(utf8_data,encoding), dialect=dialect, **kwargs)
+    csv_reader = csv.reader(utf8_data if isinstance(utf8_data,TextIOWrapper) else TextIOWrapper(utf8_data,encoding), dialect=dialect, **kwargs)
```

Comment:
this is a very long line. May I ask you to break it in few lines? Maybe use block for reformatting the file.

Reply:
Done.